src/hooks/sync-sites/use-sync-pull.ts

```diff
@@ -128,46 +137,108 @@ export function useSyncPull( {
 		[ __, client, pullStatesProgressInfo, updatePullState ]
 	);
 
+	const checkBackupFileSize = async ( downloadUrl: string ): Promise< number > => {
+		try {
+			return await getIpcApi().checkSyncBackupSize( downloadUrl );
+		} catch ( error ) {
+			console.log( 'Failed to check backup file size', error );
+			Sentry.captureException( error );
+			throw new Error( 'Failed to check backup file size' );
+		}
+	};
+
 	const onBackupCompleted = useCallback(
 		async ( remoteSiteId: number, backupState: SyncBackupState & { downloadUrl: string } ) => {
 			const { downloadUrl, selectedSite, isStaging } = backupState;
-			updatePullState( selectedSite.id, remoteSiteId, {
-				status: pullStatesProgressInfo.downloading,
-				downloadUrl,
-			} );
 
-			const filePath = await getIpcApi().downloadSyncBackup( remoteSiteId, downloadUrl );
+			try {
+				const fileSize = await checkBackupFileSize( downloadUrl );
+				console.log( 'Backup file size:', { fileSize, limit: SYNC_PUSH_SIZE_LIMIT_BYTES } );
 
-			updatePullState( selectedSite.id, remoteSiteId, {
-				status: pullStatesProgressInfo.importing,
-			} );
+				if ( fileSize > SYNC_PUSH_SIZE_LIMIT_BYTES ) {
+					console.log( 'File size exceeds limit, prompting user' );
+					const { response: userChoice } = await getIpcApi().showMessageBox( {
+						type: 'warning',
+						title: __( 'Large Backup File' ),
+						message: sprintf(
+							__(
+								'The backup file is over %s GB. You will not be able to push back the site. Continue?'
+							),
+							SYNC_PUSH_SIZE_LIMIT_BYTES
```

Comment:
I believe it should use the GB constant.

```suggestion
							SYNC_PUSH_SIZE_LIMIT_GB
```

Reply:
Good catch! Didn't realize using a modified value of constant locally.

Reply:
Thanks, looks good now.